packages/base-controller/src/BaseControllerV2.ts

```diff
@@ -147,7 +147,9 @@ export class BaseController<
   }) {
     this.messagingSystem = messenger;
     this.name = name;
-    this.#internalState = state;
+    // NOTE: We cannot assign this a type of `Immutable<...>` because Immer's
```

Comment:
Nit: Is this comment needed here? There is no type cast to explain, the type declaration is elsewhere. Nothing on this line violates expectations.

Reply:
I've clarified this comment in [bde218a](https://github.com/MetaMask/core/pull/4011/commits/bde218a8ea91d1c3a6c7871b76b841eb910ed394) so that less mistakable as a comment about a type assertion. If it hurts more than helps I'm okay with removing it though!